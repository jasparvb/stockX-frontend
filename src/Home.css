.Home .banner {
  background-repeat: no-repeat;
  height: calc(100vh - 80px);
  background-position: center;
  background-size: cover;
  background-position-y: 0%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Home h1 {
  font-size: 8rem;
  text-shadow: 0px 0px 2px black;
}

.Home .lead {
  font-size: 2rem;
  font-weight: 500;
  text-shadow: 0px 0px 2px black;
}