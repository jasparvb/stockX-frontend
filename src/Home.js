import React from "react";
import { Link } from "react-router-dom";
import BannerBg from './images/stockX-bg.jpg';
import './Home.css';

//Home page banner

function Home() {
  return (
    <section className="Home col-md-12 p-0">
      <div className="banner container-fluid text-center text-light m-0 p-5" style={{ backgroundImage: `url(${BannerBg})`}} >
        <div className="content-wrapper mb-5">
          <h1 className="mb-3 font-weight-bold">stockX</h1>
          <p className="lead">Start tracking your financial data today</p>
          <Link to="/login" className="btn btn-primary btn-lg px-5 py-3 my-5">Log In or Register</Link>
        </div>
      </div>
    </section>
  );
}

export default Home;
