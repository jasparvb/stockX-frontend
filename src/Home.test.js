import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import rootReducer from "./reducers/root";
import { createStore } from "redux";
import { BrowserRouter } from "react-router-dom";
import Home from './Home';

const store = createStore(rootReducer);

it("renders without crashing", function() {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>);
});

it("matches snapshot", function() {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  