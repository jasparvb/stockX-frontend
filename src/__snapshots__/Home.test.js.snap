// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="Home col-md-12 p-0"
  >
    <div
      class="banner container-fluid text-center text-light m-0 p-5"
      style="background-image: url(stockX-bg.jpg);"
    >
      <div
        class="content-wrapper mb-5"
      >
        <h1
          class="mb-3 font-weight-bold"
        >
          StockX
        </h1>
        <p
          class="lead"
        >
          Start tracking your financial data today
        </p>
        <a
          class="btn btn-primary btn-lg px-5 py-3 my-5"
          href="/login"
        >
          Log In or Register
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
